torch_xla/csrc/init_python_bindings.cpp

```diff
@@ -1688,8 +1688,12 @@ void InitXlaModuleBindings(py::module m) {
         [](const std::string& device_str) -> at::IValue {
           torch::lazy::BackendDevice device =
               bridge::AtenDeviceToXlaDevice(c10::Device(device_str));
-          return bridge::AtenFromXlaTensor(torch_xla::XLATensor::Create(
-              XLAGraphExecutor::Get()->GetBaseSeedData(device)));
+          at::IValue seed =
+              bridge::AtenFromXlaTensor(torch_xla::XLATensor::Create(
+                  XLAGraphExecutor::Get()->GetBaseSeedData(device)));
+          // This will update the base seed
+          XLAGraphExecutor::Get()->MarkStep(device);
```

Comment:
Is this related to the model side change, where we removed the redundant `mark_step()` calls?

Reply:
this is necessary when we remove mark_step, otherwise seed is not advancing. I will open a separate pr for this and add test.